semantic_search/main.py

```diff
@@ -71,24 +79,34 @@ def app_startup():
     model.tokenizer, model.model = setup_model_and_tokenizer(
         settings.pretrained_model_name_or_path, cuda_device=settings.cuda_device
     )
-    model.similarity = torch.nn.CosineSimilarity(-1)
+    embedding_dim = model.model.config.hidden_size
+    model.index = setup_faiss_index(embedding_dim)
 
 
 @app.post("/")
 async def query(query: Query) -> List[Dict[str, float]]:
-    ids = [query.query.uid] + [document.uid for document in query.documents]
     text = [query.query.text] + [document.text for document in query.documents]
 
     embeddings = encode(text)
```

Comment:
This is not doing what we want. It will embed each item in the query regardless of whether it already exists in the index.